src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -523,8 +565,23 @@ gl_sframe activity_classifier::perform_inference(data_iterator* data) const {
         cumulative_samples += prediction_window;
 
         // Add a row to the output SFrame.
-        writer.write({ info.session_id, preds, num_samples },
-                     /* segment_id */ 0);
+
+        if (output_frequency == "per_window") {
+
+          if (info.session_id == previous_session_id) {
```

Comment:
What happens if the `session_id` of the first prediction happens to be 1? Wouldn't we get an erroneous `prediction_id` of 2 for the first prediction then?